src/Core/Domain/Shop/ValueObject/ShopConstraint.php

```diff
@@ -94,7 +98,7 @@ public static function allShops(): self
      *
      * @throws ShopException
      */
-    public function __construct(?int $shopId, ?int $shopGroupId, bool $strict = false)
+    private function __construct(?int $shopId, ?int $shopGroupId, bool $strict = false)
```

Comment:
I think this is a BC Break

Reply:
yes, I add it in the PR description 👍

Reply:
@matthieu-rolland I just thought that this constructor should be `protected` instead of `private` This will still prevent the object from being built without the static function However it would allow creating sub types of `ShopConstraint`

For example in product page I'd like to create a `ProductShopConstraint` that would throw an exception in case you try to create it with shopGroupId (which is not handled by the product). In order to do that I need the constructor to be protected so that I can override it